lib/AST/Decl.cpp

```diff
@@ -3612,6 +3612,10 @@ void VarDecl::emitLetToVarNoteIfSimple(DeclContext *UseDC) const {
     // If the problematic decl is 'self', then we might be trying to mutate
     // a property in a non-mutating method.
     auto FD = dyn_cast_or_null<FuncDecl>(UseDC->getInnermostMethodContext());
+
+    // If this is a subscript getter, don't suggest adding mutating.
+    if (FD->getAccessorStorageDecl()->getGetter() == FD) return;
```

Comment:
Yea, I tried that, but `isImplicit` is returning false for implicit subscript get  blocks. I could try fixing that too if it should be returning true. 

The check I mentioned above though is working and I'm able to isolate case 1 cleanly now. Do the chained fixits get executed in one fixit, or does the developer have to press multiple fixit's? And do you know of where else this pattern is used? The later doesn't sound like it would be worth adding.

The existing fixit works fine for case 2 now that I'm isolating case 1 correctly.